cmd/attach/policy/attach_policy_service.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright (c) 2024 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package policy
+
+import (
+	"fmt"
+
+	"slices"
+
+	awsutil "github.com/aws/aws-sdk-go-v2/aws"
+
+	"github.com/openshift/rosa/pkg/aws"
+	"github.com/openshift/rosa/pkg/aws/tags"
+	"github.com/openshift/rosa/pkg/ocm"
+)
+
+const (
+	quotaCode = "L-0DA4ABF3"
+)
+
+type AttachPolicyService interface {
+	ValidateOptions(roleName string, policyArns []string) error
+	AutoAttachArbitraryPolicy(roleName string, policyArns []string, accountID, orgID string) (string, error)
+	ManualAttachArbitraryPolicy(roleName string, policyArns []string, accountID, orgID string) string
+}
+
+type policyService struct {
+	OCMClient *ocm.Client
+	AWSClient aws.Client
+}
+
+func NewAttachPolicyService(OCMClient *ocm.Client, AWSClient aws.Client) AttachPolicyService {
+	return &policyService{
+		OCMClient: OCMClient,
+		AWSClient: AWSClient,
+	}
+}
+
+// ValidateOptions validate the options:
+// verify rolename and each policy arn are valid
+// verify role is RedHat managed
+// verify role has quota to attach the policies
+func (p *policyService) ValidateOptions(roleName string, policyArns []string) error {
+	role, err := p.AWSClient.GetRoleByName(roleName)
+	if err != nil {
+		return fmt.Errorf(
+			"Failed to find the role %s: %s",
+			roleName, err,
+		)
+	}
+	isRedHatManaged := false
+	for _, tag := range role.Tags {
+		if awsutil.ToString(tag.Key) == tags.RedHatManaged &&
+			awsutil.ToString(tag.Value) == "true" {
+			isRedHatManaged = true
+			break
+		}
+	}
+	if !isRedHatManaged {
+		return fmt.Errorf("Rosa only supports attaching policy to RedHat managed role")
```

Comment:
Correcting message / for brevity: :) 
"Cannot attach policies to non-ROSA roles."

Reply:
message updated